internal/controller/controllers/bmo_utils.go

```diff
@@ -68,10 +68,20 @@ func (r *BMOUtils) GetIronicServiceURLS() (string, string, error) {
 		return "", "", err
 	}
 	ironicIP, inspectorIP, err := provisioning.GetIronicIP(r.kubeClient, provisioningInfo.Namespace, &provisioningInfo.Spec, r.osClient)
-	if err != nil || ironicIP == "" {
+	if err != nil {
 		r.log.WithError(err).Error("unable to determine Ironic's IP")
 		return "", "", err
 	}
+	if inspectorIP == "" {
```

Comment:
Is there a case where it's valid to have an empty `inspectorIP`? 
What can break because of this change?

Reply:
I don't think so. And even if it was empty we only have a chance of things working out.

I'd rather see this fail.